src/components/Maps/MapPage.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { useQuery } from 'react-apollo';
+import GoogleMapReact from 'google-map-react';
+
+import MenuAppBar from '../common/MenuAppBar';
+import InfoWindow from './InfoWindow';
+import { LocationPin, GET_PINS_BY_EVENT_ID } from '../../graphql/queries/maps';
+
+const MapPage = ({
+  match: {
+    params: { eventId },
+  },
+}: {
+  match: { params: { eventId: string } };
+}) => {
+  const defaultMap = {
+    center: { lat: 43.470846, lng: -80.538473 },
+    zoom: 11,
+  };
+
+  const getMapOptions = (maps) => {
+    return {
+      streetViewControl: false,
+      scaleControl: true,
+      fullscreenControl: false,
+      styles: [
+        {
+          featureType: 'poi.business',
+          elementType: 'labels',
+          stylers: [
+            {
+              visibility: 'off',
+            },
+          ],
+        },
+      ],
+      gestureHandling: 'greedy',
+      disableDoubleClickZoom: true,
+
+      mapTypeControl: true,
+      mapTypeId: maps.MapTypeId.ROADMAP,
+      mapTypeControlOptions: {
+        style: maps.MapTypeControlStyle.HORIZONTAL_BAR,
+        position: maps.ControlPosition.BOTTOM_CENTER,
+        mapTypeIds: [
+          maps.MapTypeId.ROADMAP,
+          maps.MapTypeId.SATELLITE,
+          maps.MapTypeId.HYBRID,
+        ],
+      },
+
+      zoomControl: true,
+      clickableIcons: false,
+    };
+  };
+
+  const { data, loading } = useQuery(GET_PINS_BY_EVENT_ID, {
+    variables: { eventId },
+  });
+
+  const pins: Array<LocationPin> = data && !loading ? data.pinsForEvent : [];
+  const [infoWindowOpen, setInfoWindowOpen] = React.useState(false);
+  const [interestPinTitle, setInterestPinTitle] = React.useState('');
+  const [interestPinLocation, setInterestPinLocation] = React.useState('');
+
+  const initMaps = (map, maps) => {
+    if (!loading) {
+      map.addListener('click', () => {
+        setInfoWindowOpen(false);
+        setInterestPinLocation('');
+        setInterestPinTitle('');
+      });
+      pins.map((pin) => {
+        const marker = new maps.Marker({
+          position: { lat: pin.latitude, lng: pin.longitude },
+          map,
+          title: pin.label,
+        });
+
+        marker.addListener('click', () => {
+          map.setCenter(marker.getPosition());
+          setInfoWindowOpen(true);
+          setInterestPinLocation(pin.address);
+          setInterestPinTitle(pin.label);
+        });
```

Comment:
In general, the `click` event works on desktop, but on an ipad it might not work. It doesn't work when I use the ipad pro via chrome inspection. You may have to try a touch event: https://reactjs.org/docs/events.html#touch-events.

Reply:
This part of the code is not React code, this is code from Google Maps. For Google Maps, they recommend using the 'click' listener. I tested it out on the iPad and it works well.